### ISSUE TITLE
Add standard numeric format for octal integers

### ISSUE THREAD

**Issue by colejohnson66** [dotnet/runtime#98618]
Similar to `x` and `b` for hexadecimal and binary (respectively), `o` should be provided to format "integral types" as octal.

**Comment by huoyaoyuan** [comment]
I don't think octal integers is widely used as binary/hexadecimal.

**Comment by skyoxZ** [comment]
`Convert.ToString(value, 8)` can do that.
https://learn.microsoft.com/dotnet/api/system.convert.tostring?view=net-8.0#system-convert-tostring(system-int32-system-int32)

**Comment by colejohnson66** [comment]
Octal integers are indeed less common than hexadecimal or even binary, but they're still valid numbers, and I think C# should provide more support for them. In addition, while `Convert.ToString` would work, it incurs an unnecessary allocation if I'm trying to write to a span or an ISH. It's my understanding anyway that the `Convert` class is "soft-deprecated".

Ultimately, I'd like to be able to work with spans:

```cs
int value = ...;
bool ok = value.TryFormat(buffer, out int written, "o", CultureInfo.InvariantCulture);
```

**Comment by tannergooding** [comment]
We can't add support for all number bases. Even for `binary` we only just added the support and that's after the language having had binary literals for several years.

I think we need more information showing that having dedicated octal support is worthwhile, including what spaces (outside of Unix file permissions, for example) that it is needed.

As it stands, the usage scenarios are getting sparse enough that it seems like something which should either be its own general purpose API.

**Comment by colejohnson66** [comment]
Of course, you can't add support for all bases, but octal isn't some "random" number base. I'd wager it's the most common "unsupported" number base in programming.

I'd also be fine with an API that would work with any number base. But the one that does exist (`Convert.ToString`) throws if the base isn't 2, 8, 10, or 16, and doesn't support `CultureInfo` or spans.

There's clearly a gap in the API. But you do have a valid point about usage being relatively uncommon.

**Comment by tannergooding** [comment]
This isn't something we're interested in providing at this time. Developers can provide their own extension functionality if they need it for their scenario.
